lona/html/data_binding/select.py

```diff
@@ -80,73 +154,109 @@ def readonly(self, new_value):
         else:
             del self.attributes['readonly']
 
-    # values
+    # option properties #######################################################
+    # options
     @property
-    def values(self):
+    def options(self):
         with self.lock:
-            values = []
+            options = ()
 
             for node in self.nodes:
                 if node.tag_name != 'option':
                     continue
 
-                values.append(
-                    (node.attributes.get('value', ''),
-                     node.get_text(),
-                     'selected' in node.attributes),
-                )
+                options += (node, )
 
-            return values
+            return options
 
-    @values.setter
-    def values(self, new_values):
+    @options.setter
+    def options(self, new_options):
         with self.lock:
-            self.clear()
-
-            for i in new_values:
-                value, name, selected = (list(i) + [False])[0:3]
-                option_node = Option(str(name), value=str(value))
+            self.nodes.clear()
 
-                if selected:
-                    option_node.attributes['selected'] = ''
+            if not isinstance(new_options, (list, tuple)):
+                new_options = [new_options]
 
-                self.append(option_node)
+            for option in new_options:
+                self.add_option(option)
 
-    # value
+    # selected options
     @property
-    def value(self):
+    def selected_options(self):
         with self.lock:
-            value = []
+            options = ()
+
+            for option in self.options:
```

Comment:
Let's optimize a bit: `self.options` may be calculated only once in this method.

Reply:
Ack, Fixed!